### PR TITLE
ACS-8352 Migrate to Docker Compose v2

### PR DESCRIPTION
**Pull request by dsibilio**
As per title.

### CODE REVIEW

**Review by aonych — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -332,8 +332,8 @@ jobs:
       - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.35.2
       - name: "Init"
         run: bash ./scripts/ci/init.sh
-      - name: "Run PostgreSQL 14.9 database"
-        run: docker-compose -f ./scripts/ci/docker-compose/docker-compose-db.yaml --profile postgres up -d
+      - name: "Run PostgreSQL 13.7 database"
```
Comment (aonych) [inline .github/workflows/ci.yml:335]: We shouldn't change the version.

**Review by aonych — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -358,8 +358,8 @@ jobs:
       - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.35.2
       - name: "Init"
         run: bash ./scripts/ci/init.sh
-      - name: "Run PostgreSQL 15.4 database"
-        run: docker-compose -f ./scripts/ci/docker-compose/docker-compose-db.yaml --profile postgres up -d
+      - name: "Run PostgreSQL 14.4 database"
```
Comment (aonych) [inline .github/workflows/ci.yml:361]: Same as above.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -545,8 +545,8 @@ jobs:
       - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.35.2
       - name: "Init"
         run: bash ./scripts/ci/init.sh
-      - name: "Run Postgres 15.4 database"
-        run: docker-compose -f ./scripts/ci/docker-compose/docker-compose.yaml --profile postgres up -d
+      - name: "Run Postgres 14.4 database"
```
Comment (aonych) [inline .github/workflows/ci.yml:548]: Same as above.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dsibilio) [reply .github/workflows/ci.yml]: Tyvm! Must have been some automated conflict resolution issues.

**Review by aonych — APPROVED** [review]

**Review by mikolajbrzezinski — APPROVED** [review]

**Review by damianujma — APPROVED** [review]

**Pull request merged**